src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -375,6 +383,89 @@ def migrate_alert_rule(
+            extra={"alert_rule_id": alert_rule.id},
+        )
+        return None
+
+    detector: Detector = alert_rule_detector.detector
+
+    try:
+        detector_state = DetectorState.objects.get(detector=detector)
+    except DetectorState.DoesNotExist:
+        logger.exception(
+            "DetectorState does not exist",
+            extra={"alert_rule_id": alert_rule.id, "detector_id": detector.id},
+        )
+        return None
+
+    updated_detector_fields: dict[str, Any] = {}
+    config = detector.config.copy()
+
+    for field, detector_field in FIELDS_TO_DETECTOR_FIELDS.items():
+        if updated_field := updated_fields.get(field):
+            updated_detector_fields[detector_field] = updated_field
+    # update config fields
+    config_fields = MetricAlertFire.detector_config_schema["properties"].keys()
+    for field in config_fields:
+        if field in updated_fields:
+            config[field] = updated_fields[field]
+    updated_detector_fields["config"] = config
+
+    # if the user updated resolve_threshold or threshold_type, then we also need to update the detector triggers
+    if "threshold_type" in updated_fields or "resolve_threshold" in updated_fields:
+        data_condition_group = detector.workflow_condition_group
+        if data_condition_group is None:
+            # this shouldn't be possible due to the way we dual write
+            logger.error(
+                "AlertRuleDetector has no associated DataConditionGroup",
+                extra={"alert_rule_id": alert_rule.id},
+            )
+            return None
+        data_conditions = DataCondition.objects.filter(condition_group=data_condition_group)
+        if "threshold_type" in updated_fields:
+            threshold_type = (
+                Condition.GREATER
+                if updated_fields["threshold_type"] == AlertRuleThresholdType.ABOVE.value
+                else Condition.LESS
+            )
+            resolve_threshold_type = (
+                Condition.LESS_OR_EQUAL
+                if updated_fields["threshold_type"] == AlertRuleThresholdType.ABOVE.value
+                else Condition.GREATER_OR_EQUAL
+            )
+            for dc in data_conditions:
+                if dc.condition_result == DetectorPriorityLevel.OK:
+                    dc.update(type=resolve_threshold_type)
+                else:
+                    dc.update(type=threshold_type)
+
+        if "resolve_threshold" in updated_fields:
+            resolve_condition = data_conditions.filter(condition_result=DetectorPriorityLevel.OK)
+            resolve_condition.update(comparison=updated_fields["resolve_threshold"])
+
+    detector.update(**updated_detector_fields)
+
+    # reset detector status, as the rule was updated
+    detector_state.update(active=False, state=DetectorPriorityLevel.OK)
+
+    # TODO: do we need to create an audit log entry here?
+    # do we need to return the detector's data conditions/DCG?
```

Comment:
re: audit log we probably should but I think we can punt that to another task as we'll need to create new audit log types entirely for the whole project

idk that we need to return the dc / dcg here since the user hitting this code path will be unaware of ACI as a whole, so it's not like we need to return this data in the api response. we will need to make a new serializer when we switch over though, so maybe then? I wouldn't worry about it right now though